dask/array/core.py

```diff
@@ -1151,6 +1151,12 @@ def imag(self):
     def conj(self):
         return conj(self)
 
+    def view(self, dtype):
+        mult = np.empty((1,), dtype=self.dtype).view(dtype=dtype).shape[0]
+        out = elemwise(lambda x, dt: x.copy().view(dt), self, dtype)
```

Comment:
Good point -- view relies on memory order -- they need to be C order if you
want to change the item size. ascontinguousarray would probably do the
trick, then.

On Wed, Nov 18, 2015 at 10:07 AM, Matthew Rocklin <notifications@github.com>
wrote:

> In dask/array/core.py
> <https://github.com/blaze/dask/pull/837#discussion_r45235962>:
>
> > @@ -1151,6 +1151,12 @@ def imag(self):
> >      def conj(self):
> >          return conj(self)
> >
> > +    def view(self, dtype):
> > +        mult = np.empty((1,), dtype=self.dtype).view(dtype=dtype).shape[0]
> > +        out = elemwise(lambda x, dt: x.copy().view(dt), self, dtype)
>
> View doesn't seem to work well on arrays that don't hold their own data
>
> In [1]: import numpy as np
>
> In [2]: x = np.arange(56).reshape((7, 8))
>
> In [3]: y = x[2:5, 2:5]
>
> In [4]: x.view('i2')
> Out[4]:
> array([[ 0,  0,  0,  0,  1,  0,  0,  0,  2,  0,  0,  0,  3,  0,  0,  0,  4,
>          0,  0,  0,  5,  0,  0,  0,  6,  0,  0,  0,  7,  0,  0,  0],
>        [ 8,  0,  0,  0,  9,  0,  0,  0, 10,  0,  0,  0, 11,  0,  0,  0, 12,
>          0,  0,  0, 13,  0,  0,  0, 14,  0,  0,  0, 15,  0,  0,  0],
>        [16,  0,  0,  0, 17,  0,  0,  0, 18,  0,  0,  0, 19,  0,  0,  0, 20,
>          0,  0,  0, 21,  0,  0,  0, 22,  0,  0,  0, 23,  0,  0,  0],
>        [24,  0,  0,  0, 25,  0,  0,  0, 26,  0,  0,  0, 27,  0,  0,  0, 28,
>          0,  0,  0, 29,  0,  0,  0, 30,  0,  0,  0, 31,  0,  0,  0],
>        [32,  0,  0,  0, 33,  0,  0,  0, 34,  0,  0,  0, 35,  0,  0,  0, 36,
>          0,  0,  0, 37,  0,  0,  0, 38,  0,  0,  0, 39,  0,  0,  0],
>        [40,  0,  0,  0, 41,  0,  0,  0, 42,  0,  0,  0, 43,  0,  0,  0, 44,
>          0,  0,  0, 45,  0,  0,  0, 46,  0,  0,  0, 47,  0,  0,  0],
>        [48,  0,  0,  0, 49,  0,  0,  0, 50,  0,  0,  0, 51,  0,  0,  0, 52,
>          0,  0,  0, 53,  0,  0,  0, 54,  0,  0,  0, 55,  0,  0,  0]], dtype=int16)
>
> In [5]: y.view('i2')---------------------------------------------------------------------------ValueError                                Traceback (most recent call last)<ipython-input-5-d1686749147b> in <module>()----> 1 y.view('i2')
> ValueError: new type not compatible with array.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/blaze/dask/pull/837/files#r45235962>.
>